### PR TITLE
Expose UI configuration state flow for testing

### PR DESCRIPTION
**Pull request by JakeWharton**

### CODE REVIEW

File: redwood-testing/src/commonTest/kotlin/app/cash/redwood/testing/ViewTreesTest.kt [hunk 1]
```diff
@@ -142,4 +145,20 @@ class ViewTreesTest {
 
     assertThat(widgetContainer.map { it.value }).isEqualTo(snapshot)
   }
+
+  @Test fun uiConfigurationWorks() = runTest {
+    TestSchemaTester {
```
Comment (JakeWharton) [inline redwood-testing/src/commonTest/kotlin/app/cash/redwood/testing/ViewTreesTest.kt:150]: I think I want to pipe the ability to specify the initial `UiConfiguration` instance through to this generated function. But I'll do that as a follow-up.

File: redwood-testing/src/commonMain/kotlin/app/cash/redwood/testing/TestRedwoodComposition.kt [hunk 2]
```diff
@@ -42,10 +43,10 @@ public fun <W : Any, S> TestRedwoodComposition(
   scope: CoroutineScope,
   provider: Widget.Provider<W>,
   container: Widget.Children<W>,
-  uiConfigurations: StateFlow<UiConfiguration>,
+  initialUiConfigurations: UiConfiguration = UiConfiguration(),
```
Comment (veyndan) [inline redwood-testing/src/commonMain/kotlin/app/cash/redwood/testing/TestRedwoodComposition.kt]: [nitpick]

```suggestion
  initialUiConfiguration: UiConfiguration = UiConfiguration(),
```

**Review by veyndan — APPROVED** [review]

File: redwood-testing/src/commonMain/kotlin/app/cash/redwood/testing/TestRedwoodComposition.kt [hunk 3]
```diff
@@ -42,10 +43,10 @@ public fun <W : Any, S> TestRedwoodComposition(
   scope: CoroutineScope,
   provider: Widget.Provider<W>,
   container: Widget.Children<W>,
-  uiConfigurations: StateFlow<UiConfiguration>,
+  initialUiConfiguration: UiConfiguration = UiConfiguration(),
   createSnapshot: () -> S,
 ): TestRedwoodComposition<S> {
-  return RealTestRedwoodComposition(scope, provider, container, uiConfigurations, createSnapshot)
+  return RealTestRedwoodComposition(scope, provider, container, initialUiConfigurations, createSnapshot)
```
Comment (JakeWharton) [inline redwood-testing/src/commonMain/kotlin/app/cash/redwood/testing/TestRedwoodComposition.kt]:
```suggestion
  return RealTestRedwoodComposition(scope, provider, container, initialUiConfiguration, createSnapshot)
```

File: redwood-testing/src/commonMain/kotlin/app/cash/redwood/testing/TestRedwoodComposition.kt [hunk 4]
```diff
@@ -55,14 +56,20 @@ public interface TestRedwoodComposition<S> : RedwoodComposition {
...
+  public val uiConfigurations: MutableStateFlow<UiConfiguration>
 }
 
 /** Performs Redwood composition strictly for testing. */
 private class RealTestRedwoodComposition<W : Any, S>(
   scope: CoroutineScope,
   provider: Widget.Provider<W>,
   container: Widget.Children<W>,
-  uiConfigurations: StateFlow<UiConfiguration>,
+  initialUiConfigurations: UiConfiguration,
```
Comment (JakeWharton) [inline redwood-testing/src/commonMain/kotlin/app/cash/redwood/testing/TestRedwoodComposition.kt]:
```suggestion
  initialUiConfiguration: UiConfiguration,
```

File: redwood-testing/src/commonMain/kotlin/app/cash/redwood/testing/TestRedwoodComposition.kt [hunk 5]
```diff
@@ -73,6 +80,8 @@ private class RealTestRedwoodComposition<W : Any, S>(
   /** Channel with the most recent snapshot, if any. */
   private val snapshots = Channel<S>(Channel.CONFLATED)
 
+  override val uiConfigurations = MutableStateFlow(initialUiConfigurations)
```
Comment (JakeWharton) [inline redwood-testing/src/commonMain/kotlin/app/cash/redwood/testing/TestRedwoodComposition.kt]:
```suggestion
  override val uiConfigurations = MutableStateFlow(initialUiConfiguration)
```

**Pull request merged**